app/lib/utils/gpDataMapper.js

```diff
@@ -1,6 +1,8 @@
 function getBookOnlineLink(gpData) {
-  if (gpData.bookingSystem && gpData.bookingSystem.bookOnlineLink) {
-    return gpData.bookingSystem.bookOnlineLink;
+  if (gpData.onlineServices &&
```

Comment:
The check for `gpData.onlineServices.appointments.url` isn' required.
It could be rewritten as:
```
return (gpData.onlineServices && gpData.onlineServices.appointments)
  ? gpData.onlineServices.appointments.url : undefined;
```